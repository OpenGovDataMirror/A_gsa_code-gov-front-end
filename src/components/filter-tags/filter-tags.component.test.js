import React from 'react'
import { shallow } from 'enzyme'

import FilterTags from 'components/filter-tags/filter-tags.component'

const props = {
  filters: [
    { category: 'c1', title: 't1', value: 'v1' },
    { category: 'c2', title: 't2', value: 'v2' },
  ],
  onClick: jest.fn(),
}

let wrapper
describe('components - FilterTags', () => {
  beforeEach(() => {
    wrapper = shallow(<FilterTags {...props} />)
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should map all the filters out', () => {
    expect(wrapper.findWhere(x => x.prop('value')).length).toBe(props.filters.length)
  })
})