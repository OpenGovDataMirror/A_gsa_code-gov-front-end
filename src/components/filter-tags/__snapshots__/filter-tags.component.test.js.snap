// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - FilterTags should render correctly 1`] = `
<div
  className="filter-tags"
>
  <FilterTag
    category="c1"
    key="c1:v1"
    onClick={[MockFunction]}
    title="t1"
    value="v1"
  />
  <FilterTag
    category="c2"
    key="c2:v2"
    onClick={[MockFunction]}
    title="t2"
    value="v2"
  />
</div>
`
